tests/test_integration.py

```diff
@@ -69,163 +70,205 @@ def cmd(cmd, *args, **kwargs):
     return cmd.format(*escaped_args, **escaped_kwargs)
 
 
-def _create_vm():
-
-    # Before using any functions from igvm library delete and recreate
-    # Serveradmin object.
-    vm_obj = query(servertype='vm', hostname=VM1_host)
-    try:
-        vm_obj.get()
-    except DatasetError:
-        pass
-    else:
-        vm_obj.delete()
-        vm_obj.commit()
+def setUpModule():
+    # Automatically find suitable HVs for tests.
+    # Terminate if this is impossible - we can't run tests without HVs.
+    global HVs
+    VM_route_net = Query(
+        {
+            'hostname': VM_NET,
+        },
+    ).get()['route_network']
+
+    # We can access HVs as objects but that does not mean we can compare them
+    # to any objects returned from igvm - those will be different objects,
+    # created from scratch from Serveradmin data.
+    HVs = [Hypervisor(h['hostname']) for h in Query({
+        'servertype': 'hypervisor',
+        'environment': 'testing',
+        'vlan_networks': VM_route_net,
+        'state': 'online',
+    })]
+
+    if len(HVs) < 2:
+        raise Exception('Not enough testing HVs found')
```

Comment:
I don't like that we altering the global variables from here.  Why don't we put these to the beginning of the file like the other globals?

Reply:
I don't see another way of exchanging information between setUpModule() and IGVMTest.setUpClass than a global variable.

Reply:
How about putting these to the beginning of the file like the other globals and having no `setUpModule()`?

Reply:
Having `setUpModule()` means that information about failure in preparation for tests is correctly reported to pytest. Unlike calling random code in random places with random `sys.exit()`.

Reply:
Okay then we can keep it like this until we find a better solution.  Could you please at least move `setUpModule()` function to beginning of the file?

Reply:
Done.